README.md

```diff
@@ -481,6 +485,69 @@ The verified in-toto statement may be written to stdout with the
 
 Note that `--source-uri` supports GitHub repository URIs like `github.com/$OWNER/$REPO` when the build was enabled with a Cloud Build [GitHub trigger](https://cloud.google.com/build/docs/automating-builds/github/build-repos-from-github). Otherwise, the build provenance will contain the name of the Cloud Storage bucket used to host the source files, usually of the form `gs://[PROJECT_ID]_cloudbuild/source` (see [Running build](https://cloud.google.com/build/docs/running-builds/submit-build-via-cli-api#running_builds)). We recommend using GitHub triggers in order to preserve the source provenance and valiate that the source came from an expected, version-controlled repository. You _may_ match on the fully-qualified tar like `gs://[PROJECT_ID]_cloudbuild/source/1665165360.279777-955d1904741e4bbeb3461080299e929a.tgz`.
 
+## Verification Summary Attestations (VSA)
+
+We have support for [verifying](https://slsa.dev/spec/v1.1/verification_summary#how-to-verify) VSAs.
+Rather than passing in filepaths as arguments, we allow passing in mulitple `--subject-digest` cli options, to
+accomodate subjects that are not simple-files.
+
+
+The verify-vsa command
+
+```shell
+$ slsa-verifier verify-vsa --help
+Verifies SLSA VSAs for the given subject-digests
+
+Usage:
+  slsa-verifier verify-vsa [flags] subject-digest [subject-digest...]
+
+Flags:
+      --attestations-path string      path to a file containing the attestations
+  -h, --help                          help for verify-vsa
+      --print-attestation             [optional] print the contents of attestation to stdout
+      --public-key-signing-hash-algo string   [optional] the hash algorithm used to compute the digest to be signed, one of SHA256 [default], SHA384, or SHA512
```

Comment:
The reason I had suggested this is because the key parsing libraries we use support all of Go's signature algorithms.
Thinking more about this now, maybe we can have the best of both worlds, supporting both all signature algorithms and also have a mapping between signature alg and the expected hash alg, without the need for the flag.

RSA -> Always SHA256 (RSA2048-SHA256, RSA3072-SHA256, RSA4096-SHA256 are all typical)
ECDSA-P256 -> SHA256
ECDSA-P384 -> SHA384
ECDSA-P512 -> SHA512
Ed25519 -> SHA512

This aligns with what is supported by Sigstore's spec - https://github.com/sigstore/protobuf-specs/blob/main/protos/sigstore_common.proto#L69-L86

Should be really easy to add:

```
var h crypto.Hash
switch pk := pub.(type) {
case *rsa.PublicKey:
    h = crypto.SHA256
case *ecdsa.PublicKey:
    switch pk.Curve {
    case elliptic.P256():
	    h = crypto.SHA256
    case elliptic.P384():
	    h = crypto.SHA384
    case elliptic.P521():
	    h = crypto.SHA512
    default:
	    h = crypto.SHA256 (or err out)
    }
case ed25519.PublicKey:
    h = crypto.SHA512
}
```

Thoughts? We can then drop the flag, but support all signature algorithms.

Reply:
@haydentherapper From the CLI, I think we could use your proposal, but only if the user does not specify the hashing algorithm, and then error if it can't be automatically determined by the switch-case.

Just one question: Is it uncommon to use RSA with SHA512? Unless we start having better logging, It could be a bad UX for the user if verifying the signature is failing because the hashing algorithm was incorrectly assumed by our tool to be SHA256.

Reply:
My suggestion would be to drop the flag and only configure the hash alg based on the public key.

We could later add the flag back if we find that we need to support uncommon use cases like RSA-SHA512. At least within the PKI space, it's not common. We can document what key types we support and ask users to file an issue if they need additional support.

(BTW, I missed the case of `elliptic.P224()`, so the default case of sha256 seems fine for `*ecdsa.PublicKey`).

Reply:
Ok I removed the --public-key-hash-algo option and added your switch-case

Reply:
LGTM, thanks!

Reply:
+ on the discussion. Give safe defaults and only provide an overwrite if someone complains.